src/test/util/setup_common.cpp

```diff
@@ -192,7 +196,9 @@ TestingSetup::TestingSetup(const std::string& chainName, const std::vector<const
         throw std::runtime_error(strprintf("ActivateBestChain failed. (%s)", state.ToString()));
     }
 
-    m_node.addrman = std::make_unique<AddrMan>(/* asmap */ std::vector<bool>(), /* deterministic */ false, /* consistency_check_ratio */ 0);
+    m_node.addrman = std::make_unique<AddrMan>(/* asmap */ std::vector<bool>(),
+                                               /* deterministic */ false,
+                                               gArgs.GetIntArg("-checkaddrman", 0));
```

Comment:
Seems odd to push the args into m_node.args, but then read them from gArgs. Would be clearer to use the same in both places.

Reply:
Changed the PR to avoid using `gArgs` in all places.

The global was convenient in the addrman unit tests which have 4 different AddrMan-derived classes. I think now is better because all of them receive the same arguments as the base class `AddrMan`. This increased the size of commit `test: addrman unit tests: override-able check ratio`, I think it is justified.